### PR TITLE
Feature/bal 7 gp sign up

### PR DESCRIPTION
**Pull request by gperissetcelteeka**
# Sign UP
## :recycle: Current situation & Problem
*Describe the current situation (if possible, with an exemplary (or real) code snippet and/or where this is used)*
*Please link any open issue that is addressed with this PR*
#7 

## :bulb: Proposed solution
*Describe the solution and how this affects the project and internal structure*

## :gear: Release Notes 
*Add a summary of the feature and possible migration guides if this is a breaking change so this section can be added to the release notes.*
*Include code snippets that provide examples of the feature implemented if it appends or changes the public interface.*

## :heavy_plus_sign: Additional Information
*Provide some additional information if possible*

### Related PRs
*Reference the related PRs*

### Testing
*Are there tests included? If yes, which situations are tested, and which corner cases are missing?*

### Reviewer Nudging
*Where should the reviewer start? Where is a good entry point?*

### Code of Conduct & Contributing Guidelines 

By submitting creating this pull request, you agree to follow our [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md):
- [x] I agree to follow the [Code of Conduct](https://github.com/CS342/.github/blob/main/CODE_OF_CONDUCT.md) and [Contributing Guidelines](https://github.com/CS342/.github/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: Balance/BalanceExtentions.swift
Comment (sebastianricaldoni) [inline Balance/BalanceExtentions.swift:1]: Really minor @gperissetcelteeka but how hard would it be to name it `BalanceExtensions.swift` ?

File: Balance/Login/BalanceAccount.swift [hunk 1]
```diff
@@ -20,37 +20,44 @@ struct BalanceAccount: View {
...
+    //                .font(.custom("Montserrat-Regular", size: 15))
+    //            Button {
+    //                onboardingSteps.append(.signUp)
+    //            } label: {
+    //                Text("Create an account")
+    //                    .foregroundColor(primaryColor)
+    //                    .font(.custom("Montserrat-SemiBold", size: 15))
+    //            }
+    //        }
+    //    }
```
Comment (sebastianricaldoni) [inline Balance/Login/BalanceAccount.swift:60]:
```suggestion
```

File: Balance/Login/BalanceAccount.swift [hunk 2]
```diff
@@ -20,37 +20,44 @@ struct BalanceAccount: View {
...
+            //                actionView
+            //            }
+            //        )
+            //        .onReceive(account.objectWillChange) {
+            //            if account.signedIn {
+            //                completedOnboardingFlow = true
+            //            }
         }
-        .ignoresSafeArea()
+//        .ignoresSafeArea()
```
Comment (sebastianricaldoni) [inline Balance/Login/BalanceAccount.swift:43]:
```suggestion
```

File: Balance/Login/BalanceAccount.swift [hunk 3]
```diff
@@ -20,37 +20,44 @@ struct BalanceAccount: View {
...
+            //            contentView: {
+            //                LoginView().environmentObject(AuthViewModel())
+            //            }, actionView: {
+            //                actionView
+            //            }
+            //        )
+            //        .onReceive(account.objectWillChange) {
+            //            if account.signedIn {
+            //                completedOnboardingFlow = true
+            //            }
```
Comment (sebastianricaldoni) [inline Balance/Login/BalanceAccount.swift:41]:
```suggestion
```

File: Balance/Login/SignInView.swift [hunk 4]
```diff
@@ -232,12 +255,86 @@ struct SignInView: View {
...
+        
+        if !NSPredicate(format: "SELF MATCHES %@", ".*[a-z]+.*").evaluate(with: str) {
+            errors.append("At least one lowercase")
+        }
+        
+        if str.count < 8 {
+            errors.append("Min 8 characters total")
+        }
+        return errors
+    }
```
Comment (sebastianricaldoni) [inline Balance/Login/SignInView.swift:325]: @gperissetcelteeka just a thought here because I'm not aware of the reach of CardinalKit and/or Firebase but do you know if password policy is something that can get set in the backend side? I'm just concerned about the cases where some policy changes and we need to release a new version of the app. I don't think this is a pressing need but more something to check if we have the time. Thanks!

**Review by sebastianricaldoni — APPROVED** [review]
LGTM 🚀 
Just a few minor comments and suggestions to remove comment blocks

File: Balance/Login/SignInView.swift (hunk 4 above)
Comment (gperissetcelteeka) [reply Balance/Login/SignInView.swift:325]: I have to investigate that.

File: Balance/BalanceExtentions.swift
Comment (gperissetcelteeka) [reply Balance/BalanceExtentions.swift:1]: I will change it in the next PR
